cpp/src/arrow/acero/asof_join_node.cc

```diff
@@ -370,15 +370,10 @@ struct MemoStore {
     times_.swap(memo.times_);
   }
 
-  // Updates the current time to `ts` if it is less. A different thread may win the race
-  // to update the current time to more than `ts` but not to less. Returns whether the
-  // current time was changed from its value at the beginning of this invocation.
+  // Updates the current time to `ts` if it is less. Returns true if updated.
   bool UpdateTime(OnType ts) {
-    OnType prev_time = current_time_;
```

Comment:
Are these change out of the scope of this PR? If so, can you create separate ones for them?

Reply:
Created #36094 and will revert here.